demos/http/http_demo_basic_tls/http_demo_basic_tls.c

```diff
@@ -741,35 +297,55 @@ int main( int argc,
 {
     /* Return value of main. */
     int returnStatus = EXIT_SUCCESS;
-    /* The HTTP Client library transport layer interface. */
-    HTTPTransportInterface_t transportInterface;
-    /* Structure based on the definition of the HTTP network context. */
-    struct NetworkContext networkContext;
+    /* The transport layer interface used by the HTTP Client library. */
+    TransportInterface_t transportInterface;
+    /* The network context for the transport layer interface. */
+    NetworkContext_t networkContext;
+    /* Credentials to establish the TLS connection. */
+    OpensslCredentials_t opensslCredentials;
+    /* Status returned by OpenSSL transport implementation. */
+    OpensslStatus_t opensslStatus;
+    /* Information about the server to send the HTTP requests. */
+    ServerInfo_t serverInfo;
 
     ( void ) argc;
     ( void ) argv;
 
-    /**************************** Connect. ******************************/
+    /* Initialize TLS credentials. */
+    ( void ) memset( &opensslCredentials, 0, sizeof( opensslCredentials ) );
+    opensslCredentials.pRootCaPath = ROOT_CA_CERT_PATH;
```

Comment:
The openssl API expects a NULL terminated certificate, correct?

Reply:
The underlying OpenSSL API call requires it. We used to call malloc just to produce a null-terminated string to pass to the API call but just seems like overkill.